src/Marten/Services/SessionOptions.cs

```diff
@@ -93,6 +101,23 @@ internal IConnectionLifetime Initialize(DocumentStore store, CommandRunnerMode m
             throw new DefaultTenantUsageDisabledException();
         }
 
+        var innerConnectionLifetime =GetInnerConnectionLifetime(store, mode);
+
+        var corelationId = Activity.Current is not null ? Activity.Current.RootId : Guid.NewGuid().ToString();
+        var tags = new Dictionary<string, object>
+        {
+            { MartenCorelationId, corelationId },
+            { MartenTenantId, Tenant.TenantId }
+        };
+
+        return OpenTelemetryOptions.TrackConnectionEvents
+            ? new EventTracingConnectionLifetime(innerConnectionLifetime, OpenTelemetryOptions,
+                StartConnectionActivity(Activity.Current, tags))
+            : innerConnectionLifetime;
```

Comment:
Ideally we only allocate the event tracing lifetime if there's any active listeners:
```suggestion
        return (OpenTelemetryOptions.TrackConnectionEvents && ActivitySource.HasListeners())
            ? new EventTracingConnectionLifetime(innerConnectionLifetime, OpenTelemetryOptions,
                StartConnectionActivity(Activity.Current, tags))
            : innerConnectionLifetime;
```